pkg/driver/driver_azure.go

```diff
@@ -329,6 +338,14 @@ func (d *AzureDriver) createVMNicDisk() (*compute.VirtualMachine, error) {
 		nicName           = dependencyNameFromVMName(vmName, nicSuffix)
 	)
 
+	if d.AzureMachineClass.Spec.Properties.Zone == nil && d.AzureMachineClass.Spec.Properties.AvailabilitySet == nil {
+		return nil, errors.New("Machine need to be assigned to AvailabilityZone or an AvailabilitySet")
+	}
+
+	if d.AzureMachineClass.Spec.Properties.Zone != nil && d.AzureMachineClass.Spec.Properties.AvailabilitySet != nil {
+		return nil, errors.New("Machines assigned to AvailabilityZone and AvailabilitySet in parallel are not supported")
+	}
+
```

Comment:
These checks should be moved to `validation` package [here](https://github.com/gardener/machine-controller-manager/blob/rel-0.21.0/pkg/apis/machine/validation/azuremachineclass.go)

Reply:
Thanks wasn't aware of that. I will move the checks to the validation.

Reply:
I have updated the PR.
Unfortunately it is not possible to raise an error which reference multiple fields therefore I note `zone` and `availabilitySet` in the same field reference. Is that ok or do prefer a dedicated error per field? Which mean we would get two validation errors for one error situation.